pyanaconda/modules/storage/devicetree/model.py

```diff
@@ -266,15 +273,17 @@ def reset(self, cleanup_only=False):
 
         # Protect devices from teardown.
         self._mark_protected_devices()
-        self.devicetree.teardown_all()
+        if deep:
+            self.devicetree.teardown_all()
 
         self.fsset = FSSet(self.devicetree)
 
         # Clear out attributes that refer to devices that are no longer in the tree.
         self.bootloader.reset()
 
         self.roots = []
-        self.roots = find_existing_installations(self.devicetree)
+        if deep:
+            self.roots = find_existing_installations(self.devicetree)
```

Comment:
I talked to Vojta and we think this can be removed from the `reset` method. It can be called directly via the `FindExistingSystemsWithTask` DBus method. We can call it in the `reset_storage` function by default to handle GUI, TUI and the rescue mode and Web UI can handle this on its own. What do you think?